packages/ra-ui-materialui/src/field/TranslatableFields.tsx

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * Provides a way to show multiple languages for any fields passed as children.
+ *
+ * @example <caption>Basic usage</caption>
+ * <TranslatableFields locales={['en', 'fr']}>
+ *     <TextField source={getSource('title')} />
+ *     <TextField source={getSource('description')} />
+ * </TranslatableFields>
+ *
+ * @example <caption>With a custom language selector</caption>
+ * <TranslatableFields
+ *     selector={<MyLanguageSelector />}
+ *     locales={['en', 'fr']}
+ * >
+ *     <TextField source={getSource('title')} />
+ * <TranslatableFields>
+>
+ *
+ * const MyLanguageSelector = () => {
+ *     const {
+ *         locales,
+ *         selectedLocale,
+ *         selectLocale,
+ *     } = useTranslatableContext();
+ *
+ *     return (
+ *         <select onChange={selectLocale}>
+ *             {locales.map((locale) => (
+ *                 <option selected={locale.locale === selectedLocale}>
+ *                     {locale.name}
+ *                 </option>
+ *             ))}
+ *        </select>
+ *     );
+ * }
+ *
+ * * @param props The component props
+ * * @param {string} props.defaultLocale The locale selected by default. Default to 'en'.
+ * * @param {string[]} props.locales An array of the possible locales in the form. For example [{ 'en', 'fr' }].
+ * * @param {ReactElement} props.selector The element responsible for selecting a locale. Defaults to Material UI tabs.
+ */
+export const TranslatableFields = (
+    props: TranslatableFieldsProps
+): ReactElement => {
+    const {
+        defaultLocale,
+        locales,
+        groupKey = '',
+        selector = <TranslatableFieldsTabs groupKey={groupKey} />,
+        children,
+        record,
+        resource,
+        basePath,
+    } = props;
+    const context = useTranslatable({ defaultLocale, locales });
+    const classes = useStyles(props);
+
+    return (
+        <div className={classes.root}>
+            <TranslatableContextProvider value={context}>
+                {selector}
+                {locales.map(locale => (
+                    <TranslatableFieldsTabContent
+                        key={locale}
+                        basePath={basePath}
+                        locale={locale}
+                        record={record}
+                        resource={resource}
+                        groupKey={groupKey}
+                    >
+                        {children}
+                    </TranslatableFieldsTabContent>
+                ))}
+            </TranslatableContextProvider>
+        </div>
+    );
+};
+
+export interface TranslatableFieldsProps extends UseTranslatableOptions {
+    basePath: string;
+    children: ReactNode;
+    classes?: ClassesOverride<typeof useStyles>;
+    record: Record;
+    resource: string;
+    selector?: ReactElement;
+    groupKey?: string;
+}
+
+const useStyles = makeStyles(theme => ({
+    root: {
+        flexGrow: 1,
+        backgroundColor: theme.palette.background.default,
```

Comment:
why don't you use the material-ui default style for tabs?

Reply:
There is no such thing. Material-ui only provides tabs but no tab panels. Our fields and inputs are used in views which define a white background by default. I chose to set a different background for the tab panels so that the hierarchy is clearer

Reply:
Sorry, my mistake. I thought that's where you added the blue background to the tab headers, but I realize it's mui's default. 

that being said, I think the default in this situation should be a white background (as in FormTabs), with the ability for the developer to override it. 

WDYT?

Reply:
I think that with the same background as the form, users won't be able to see which inputs are tied to the translation tabs and which aren't. We should probably make it easier for them by default